### ISSUE TITLE
stack-error too deep on ActiveRecord#save w/db monitoring agent

### ISSUE THREAD

**Issue by jakewilkins** [scoutapp/scout_apm_ruby#127]
Hello!

We hit a snag deploying `2.3.0.pre` to production. It looks like both gems are hooking into ActiveRecord#save in different ways and it's causing it.

```
  File "/app/vendor/bundle/ruby/2.4.0/gems/scout_apm-2.3.0.pre/lib/scout_apm/instruments/active_record.rb", line 266, in save_with_scout_instruments
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent/instrumentation/active_record_5.rb", line 15, in block in save
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent/transaction.rb", line 859, in with_database_metric_name
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent.rb", line 587, in with_database_metric_name
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent/instrumentation/active_record_5.rb", line 14, in save
  File "/app/vendor/bundle/ruby/2.4.0/gems/scout_apm-2.3.0.pre/lib/scout_apm/instruments/active_record.rb", line 266, in save_with_scout_instruments
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent/instrumentation/active_record_5.rb", line 15, in block in save
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent/transaction.rb", line 859, in with_database_metric_name
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent.rb", line 587, in with_database_metric_name
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent/instrumentation/active_record_5.rb", line 14, in save
  File "/app/vendor/bundle/ruby/2.4.0/gems/scout_apm-2.3.0.pre/lib/scout_apm/instruments/active_record.rb", line 266, in save_with_scout_instruments
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent/instrumentation/active_record_5.rb", line 15, in block in save
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent/transaction.rb", line 859, in with_database_metric_name
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent.rb", line 587, in with_database_metric_name
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent/instrumentation/active_record_5.rb", line 14, in save
  File "/app/vendor/bundle/ruby/2.4.0/gems/scout_apm-2.3.0.pre/lib/scout_apm/instruments/active_record.rb", line 266, in save_with_scout_instruments
  File "/app/vendor/bundle/ruby/2.4.0/gems/newrelic_rpm-4.4.0.336/lib/new_relic/agent/instrumentation/active_record_5.rb", line 15, in block in save
.... many many many more lines of this same thing.
```

As best I can tell, [this line in this commit](https://github.com/scoutapp/scout_apm_ruby/blame/5bf862f87eaee055bff6d3ba4598c1c50c20e409/lib/scout_apm/instruments/active_record.rb#L266) using `alias_method` chaining really doesn't work well when combined with the [prepending](https://github.com/newrelic/rpm/blob/6027802e8100f66012cf6b1bd43c6d1b5c8aeaaf/lib/new_relic/agent/instrumentation/active_record_5.rb#L88) that NewRelic does.

We rolled the `scout_apm` back to before that commit and it's working fine again, but since we were testing out the `pre` I thought I'd put up a flag for this.

💕 💖

**Comment by ewalk153** [comment]
I’m having the same issue.

**Comment by cschneid** [comment]
Released 2.3.0.pre2 which should prevent the interaction with NewRelic from causing this issue. Another solution appears to be flipping the `Gemfile` order of NewRelic & Scout.

**Comment by cschneid** [comment]
Closing this - since we've not heard any reports of continued issues. Please let us know if this comes up again.
